### PR TITLE
Allow configurable resource requests and limits through command arguments.

### PR DESCRIPTION
**Pull request by musa-asad**
*Description of changes:*
Allow configurable resource limits through command arguments so operator can take these values and use them for init containers.

*Testing:*
See https://github.com/aws/amazon-cloudwatch-agent-operator/pull/196.

1. With these local helm chart changes, I ran `helm upgrade --install amazon-cloudwatch-observability helm-charts/charts/amazon-cloudwatch-observability --values helm-charts/charts/amazon-cloudwatch-observability/values.yaml --set clusterName=cloudwatchagent-operator --set region=us-west-2`.
2. Then, ran `kubectl get pods`.
3. Then, ran `kubectl describe pod amazon-cloudwatch-observability-controller-manager-####`.

As we can see, it has the new operator image and command arguments.
Screenshot 2024-08-09 at 3 53 26 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: charts/amazon-cloudwatch-observability/values.yaml [hunk 1]
```diff
@@ -446,6 +446,13 @@ manager:
   service:
     name:
 
+## Default resource limits for init containers.
+initContainers:
+  resources:
+    limits:
+      cpu: 100m
+      memory: 64Mi
```
Comment (musa-asad) [inline charts/amazon-cloudwatch-observability/values.yaml:454]: This is a placeholder default value, subject to change.

**Review by lisguo — APPROVED** [review]

File: charts/amazon-cloudwatch-observability/values.yaml [hunk 2]
```diff
@@ -410,14 +410,35 @@ manager:
       repositoryDomain: public.ecr.aws/aws-observability
       repository: adot-autoinstrumentation-java
       tag: v1.32.3
+      resources:
```
Comment (sky333999) [inline charts/amazon-cloudwatch-observability/values.yaml:413]: These are not related to the image.

Comment (musa-asad) [reply charts/amazon-cloudwatch-observability/values.yaml:413]: To assign different resource requests and limits per image, is that okay?

**Review by Paramadon — APPROVED** [review]

File: charts/amazon-cloudwatch-observability/values.yaml (hunk 2 above)
Comment (musa-asad) [reply charts/amazon-cloudwatch-observability/values.yaml:413]: Being addressed in https://github.com/aws-observability/helm-charts/pull/80.
